hashing/src/indexed_merkle_proof.rs

```diff
@@ -0,0 +1,465 @@
+use itertools::Itertools;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use casper_types::bytesrepr::{self, FromBytes, ToBytes};
+
+use crate::{
+    error::{MerkleConstructionError, MerkleVerificationError},
+    Digest,
+};
+
+#[derive(PartialEq, Debug, JsonSchema, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct IndexedMerkleProof {
+    index: u64,
+    count: u64,
+    merkle_proof: Vec<Digest>,
+}
+
+impl ToBytes for IndexedMerkleProof {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        let mut result = bytesrepr::allocate_buffer(self)?;
+        result.append(&mut self.index.to_bytes()?);
+        result.append(&mut self.count.to_bytes()?);
+        result.append(&mut self.merkle_proof.to_bytes()?);
+        Ok(result)
+    }
+
+    fn serialized_length(&self) -> usize {
+        self.index.serialized_length()
+            + self.count.serialized_length()
+            + self.merkle_proof.serialized_length()
+    }
+}
+
+impl FromBytes for IndexedMerkleProof {
+    fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
+        let ((index, count, merkle_proof), remainder) =
+            <(u64, u64, Vec<Digest>)>::from_bytes(bytes)?;
```

Comment:
Let's be consistent in usage of tuples in `to_bytes()` and `from_bytes()` please?

Reply:
Fixed in https://github.com/casper-network/casper-node/pull/2119/commits/c28bba0adc60c4c5116381e5d4d95c5ca295ddbf